tsconfig.json

```diff
@@ -1,42 +1,35 @@
 {
+  "extends": "@canonical/typescript-config-react",
   "compilerOptions": {
     "outDir": "./public/js/dist",
     "baseUrl": ".",
     "paths": {
       "api/*": ["src/api/*"],
       "context/*": ["src/context/*"],
       "components/*": ["src/components/*"],
+      "lib/*": ["src/lib/*"],
       "pages/*": ["src/pages/*"],
       "types/*": ["src/types/*"],
       "util/*": ["src/util/*"],
       "*": ["types/*.d.ts"]
     },
-    "target": "ES2020",
     "module": "esnext",
-    "jsx": "react-jsx",
-    "strict": true,
+    "moduleResolution": "node",
     "esModuleInterop": true,
-    "skipLibCheck": true,
-    "forceConsistentCasingInFileNames": true,
-    "lib": ["dom", "dom.iterable", "esnext"],
-    "allowJs": true,
     "allowSyntheticDefaultImports": true,
-    "noFallthroughCasesInSwitch": true,
-    "moduleResolution": "node",
-    "resolveJsonModule": true,
-    "isolatedModules": true,
```

Comment:
I think we should have `isolatedModules` set to `true` still, since we are using `vite` which uses `esbuild` under the hood for transpilation (see [reference](https://v2.vitejs.dev/guide/features.html#typescript-compiler-options))